python_modules/dagster/dagster/core/host_representation/external.py

```diff
@@ -490,6 +496,12 @@ def get_external_origin(self):
     def get_external_origin_id(self):
         return self.get_external_origin().get_id()
 
+    @property
+    def selector_id(self):
+        if self._selector_id is None:
+            self._selector_id = create_snapshot_id(self._selector)
+        return self._selector_id
```

Comment:
is this expensive enough that this cache is useful?

Reply:
perhaps not... i guess I just trained myself to be wary of it.  I can remove for simplicity and add back if necessary.